File: backend/src/main/java/com/urmyfood/backend/domain/model/Order.java
package com.urmyfood.backend.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order {
    private UUID orderId;
    private Account customer;
    private Account shop;
    private Voucher voucher;
    private BigDecimal totalAmount;
    private BigDecimal discountAmount;
    private BigDecimal finalAmount;
    private OrderStatus orderStatus;
    private PaymentMethod paymentMethod;
    private PaymentStatus paymentStatus;
    private String deliveryAddress;
    private String note;
    private String cancelReason;
    private List<OrderItem> items;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}

File: frontend/app-shop/src/main/java/com/urmyfood/shop/domain/model/Order.kt
package com.urmyfood.shop.domain.model

data class Order(
    val orderId: String,
    val customerId: Long,
    val customerName: String,
    val customerPhone: String,
    val shopId: Long,
    val totalAmount: Double,
    val discountAmount: Double,
    val finalAmount: Double,
    val orderStatus: String,
    val paymentMethod: String,
    val paymentStatus: String,
    val deliveryAddress: String,
    val note: String?,
    val cancelReason: String?,
    val createdAt: String,
    val items: List<OrderItem>
)
